src/components/views/messages/MPollBody.tsx

```diff
@@ -283,13 +439,58 @@ export default class MPollBody extends React.Component<IBodyProps, IState> {
     }
 }
 
+function renderEndedOption(
+    answer: IPollAnswer,
+    checked: boolean,
+    votesText: string,
+) {
+    const classNames = `mx_MPollBody_endedOption${
+        checked ? " mx_MPollBody_endedOptionWinner": ""
+    }`;
+
+    return <div className={classNames} data-value={answer.id}>
+        <div className="mx_MPollBody_optionDescription">
+            <div className="mx_MPollBody_optionText">
+                { answer[TEXT_NODE_TYPE.name] }
+            </div>
+            <div className="mx_MPollBody_optionVoteCount">
+                { votesText }
+            </div>
+        </div>
+    </div>;
+}
+
+function renderOption(
+    pollId: string,
+    answer: IPollAnswer,
+    checked: boolean,
+    votesText: string,
+    onOptionSelected: (e: React.FormEvent<HTMLInputElement>) => void,
+) {
+    return <StyledRadioButton
+        name={`poll_answer_select-${pollId}`}
+        value={answer.id}
+        checked={checked}
+        onChange={onOptionSelected}
+    >
+        <div className="mx_MPollBody_optionDescription">
+            <div className="mx_MPollBody_optionText">
+                { answer[TEXT_NODE_TYPE.name] }
+            </div>
+            <div className="mx_MPollBody_optionVoteCount">
+                { votesText }
+            </div>
+        </div>
+    </StyledRadioButton>;
+}
```

Comment:
These should be refactored into Functional components to be more akin to React patterns.

PascalCased name, props should be an object, and change callee to instantiate using JSX.

Reply:
Thank you for the detailed review, and sorry there was a lot to fix!

Reply:
Had a go at this in f90b28d398